scripts/globals/mobskills/ultimate_terror.lua

```diff
@@ -0,0 +1,61 @@
+---------------------------------------------
+-- Ultimate Terror
+--
+-- Description: AoE Absorb All with randomness
+-- Type: Magical
+---------------------------------------------
+require("scripts/globals/settings")
+require("scripts/globals/status")
+require("scripts/globals/mobskills")
+---------------------------------------------
+local mobskill_object = {}
+
+mobskill_object.onMobSkillCheck = function(target, mob, skill)
+    return 0
+end
+
+mobskill_object.onMobWeaponSkill = function(target, mob, skill)
+
+    local drained = 0
+    local threshold = 3/7
+
+    if math.random() < threshold then
+        skill:setMsg(xi.mobskills.mobDrainAttribute(mob, target, xi.effect.STR_DOWN, 10, 3, 60))
```

Comment:
You could put these effects into a table and use a for loop to move through it rather than having this many if statements:
```suggestion
        local effects = 
        {
          xi.effect.STR_DOWN,
          xi.effect.DEX_DOWN,
        }
```
etc.

Reply:
I usually don't mind sequential if statements when there are only a few, but info agree this is kind of ugly.  It's broken out in case there were any differences in power required for each debuff, but if they are all the same, then I'd agree that the for loop with a single if statement is probably better.

Reply:
I see what you mean! If it's the case that they're all unique, this is probably not a bad way to implement. but if it's just a couple of the effects that are different you could add a few if statements within the for loop for those particular cases.

Reply:
Let's get this changed to the loop just for stylistic purposes and we will be good to go.

Reply:
Already done.